tt_eager/tt_dnn/op_library/bmm/multi_core_reuse_mcast_2d_optimized/bmm_op_multi_core_reuse_mcast_2d_optimized.cpp

```diff
@@ -69,7 +71,24 @@ operation::ProgramWithCallbacks create_program_mcast_in0_in1(
     uint32_t out_CB_size = out_CB_tiles * output_single_tile_size;
     uint32_t interm0_CB_size = out_CB_tiles * interm0_single_tile_size;
 
-    uint32_t in2_block_tiles = per_core_M * in0_block_w;
+    uint32_t in2_block_tiles;
+    uint32_t in0_shard_width_in_tiles = 0;
+    uint32_t in0_shard_height_in_tiles = 0;
+    if (in0_is_sharded) {
+        in0_shard_width_in_tiles = in0_buffer->shard_spec().shape()[1] / 32;
+        in0_shard_height_in_tiles = in0_buffer->shard_spec().shape()[0] / 32;
+        in2_block_tiles = per_core_M * in0_shard_width_in_tiles;
+    } else {
+        in2_block_tiles = per_core_M * in0_block_w;
+    }
```

Comment:
If in0 is interleaved, cb2 isn't used I think?

Reply:
right...